flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_AirwaysFromWaypointPage.js

```diff
@@ -79,21 +86,22 @@ class A320_Neo_CDU_AirwaysFromWaypointPage {
                         subRows[i + 1] = ["\xa0VIA", ""];
                         mcdu.onLeftInput[i + 1] = async (value) => {
                             if (value.length > 0) {
-                                const toWp = await this._getFirstIntersection(mcdu.flightPlanManager, value, pendingAirway.icaos);
+                                const toWp = await this._getFirstIntersection(fmc.flightPlanManager, value, pendingAirway.icaos);
                                 if (toWp) {
-                                    mcdu.ensureCurrentFlightPlanIsTemporary(() => {
-                                        mcdu.insertWaypointsAlongAirway(toWp, mcdu.flightPlanManager.getEnRouteWaypointsLastIndex() + 1, pendingAirway.name, async (result) => {
+                                    fmc.ensureCurrentFlightPlanIsTemporary(() => {
+                                        fmc.insertWaypointsAlongAirway(toWp, fmc.flightPlanManager.getEnRouteWaypointsLastIndex() + 1, pendingAirway.name, async (result) => {
                                             if (result) {
-                                                const airway = await this._getAirway(mcdu, value);
+                                                const airway = await this._getAirway(fmc, value);
                                                 if (airway) {
-                                                    A320_Neo_CDU_AirwaysFromWaypointPage.ShowPage(mcdu, waypoint, offset, airway);
+                                                    A320_Neo_CDU_AirwaysFromWaypointPage.ShowPage(fmc, mcdu, waypoint, offset, airway);
                                                 } else {
                                                     mcdu.addNewMessage(NXSystemMessages.noIntersectionFound);
                                                 }
                                             } else {
                                                 mcdu.addNewMessage(NXSystemMessages.noIntersectionFound);
                                             }
                                         });
+                                        mcdu.requestOffsideUpdate();
```

Comment:
**issue**: `mcdu.requestOffsideUpdate();` may execute before `fmc.insertWaypointsAlongAirway` calls the callback, due to the asynchronous nature of `fmc.insertWaypointsAlongAirway`.
**suggestion**: Either await `insertWaypointsAlongAirway` or move the `requestOffsideUpdate` into the callback.